### ISSUE TITLE
Can we have more than one endpoint for one infores?

### ISSUE THREAD

**Issue by edeutsch** [NCATSTranslator/TranslatorArchitecture#93]
ARAX code had made the assumption that there would not be two endpoints (at the same TRAPI version) hosting different data advertising the same infores key. When this happened, ARAX ended up dropping one:

![image](https://github.com/NCATSTranslator/TranslatorArchitecture/assets/12707718/59ddbc37-41c7-4b6f-983e-961ee5d20bb8)

Should it be permitted to have two different endpoints providing different information (and thus both be considered), both advertising the same infores key?

This may have downstream use implications.

**Comment by cbizon** [comment]
We previously decided that this would be ok.  See: https://github.com/NCATSTranslator/TranslatorArchitecture/blob/master/SmartAPIRegistration.md#server-count

We could always revisit this decision, but currently it is explicitly allowed.

**Comment by cbizon** [comment]
To be clear though, there should only be 1 smartapi registration, it is allowed to have two servers in the server block for one environment.

**Comment by edeutsch** [comment]
@cbizon I don't think these answer my question, but I suppose my question wasn't very clear. I have changed the title. The question really is: Can there be two endpoints with different SmartAPI registrations providing completely different data using the same infores?

The URL that you quote above says this:
> There must exist at least one server per (infores, x-maturity). It is allowed to register two or more server urls for a given (infores, x-maturity), for instance to provide both http and https interfaces. The caller to these services may use either and they must provide equivalent responses.

Notably, "they must provide equivalent responses".

The endpoints from my question provide completely different responses.

**Comment by cbizon** [comment]
Gotcha.  That's right.  These look like different services, and should therefore have different infores identifiers.  Maybe the first should be "infores:gelinea" or similar.

**Comment by cbizon** [comment]
@vdancik ?

**Comment by vdancik** [comment]
I did change the SmartAPI registration when I was alerted the issue, but I think that as a matter of policy it makes sense to allow multiple SmartAPI registrations per infores.

**Comment by gglusman** [comment]
@vdancik How so? It seems to lead to confusion...

**Comment by RichardBruskiewich** [comment]
Short answer to this issue: no. Infores usage in the SmartAPI Registry should be globally unique as to the cross-product of infores and TRAPI version (however, the Biolink Model release version is not constraining here, I believe... @sierra-moxon?)

**Comment by karafecho** [comment]
I agree that infores id's should be globally unique. I'll add that having two identical infores id's in the infores catalog, with each pointing to distinct services will present a major headache for the UI team, as they rely heavily on the infores catalog for finding resources and associated information (e.g., wiki URLs).

**Comment by sierra-moxon** [comment]
If the data is completely different, we need a new infores identifier.  The version of the resource is not included in the definition of unique for infores identifiers.
